snippets/MappedList.md

```diff
@@ -0,0 +1,40 @@
+### MappedList
+
+Renders a list of elements from an array of data.
+
+Use the value of the `isOrderedList` prop to conditionally render a `<ol>` or `<ul>` list.
+Use `Array.prototype.map` to render every item in `data` as a `<li>` element and give it a `key`.
+`data` can either be an array of objects with the `id` and `value` properties or an array of primitives.
+Omit the `isOrderedList` prop to render a `<ul>` list by default.
+
+```jsx
+function MappedList(props) {
+  return props.isOrderedList ? (
+    <ol>
+      {props.data.map((v, i) => (
```

Comment:
Are we sure we don't want to allow a format prop to be passed? Basically making this take three props:

- `isOrderedList` which designates surrounding tag
- `format` which gets passed to map
- `data` which is self explanatory

Basically this now looks like a call to map on an array of data

```js
<MappedList isOrderedList={true} data={['a','b','c']} format={(value, index) => <li key={index}>value</li>} />
```

As for a default formatter, we can just add it right before the return statement

```js
const format = props.format || (value, index) => (<li key={index}>value</li>)
```

then each map becomes

```js
const list = props.data.map(format);
```

then the return value is

```js
props.isOrderedList ? <ol>{list}</ol> : <ul>{list}</ul>
```

Either way we should also make the mapping happen before the return so we don't have two anonymous functions.

Reply:
The issue with the formatter is that we cannot enforce keying if the user passes a function that does not key the data properly. I would suggest we leave it as-is. The rest of the suggestions are being implemented as we speak.

Reply:
Although then that would warrant a note... mapped items are annoying, not gonna lie

Reply:
I totally missed that one. Is formatter in the scope of a beginner snippet like this or should it be an advanced version like we are discussing under #9?

Reply:
Considering its an argument to map, maybe we call it `map`? The name is not really important but the idea is. For this one it's as advanced as `Array#map()`... so yeah